docs(nx-dev): add personal access token blog post (#27719)
nrwl/nx: docs/blog/2024-09-05-personal-access-tokens.md

## Access Tokens and the problem of revoking access

Our previous implementation of access tokens required you to commit the access token to your `nx.json` file. Typically,
service providers don't recommend committing any kind of API token like this, and we've fielded a lot of questions about
this practice in the past. It is safe to commit this token, as access to the cached artifacts of Nx Cloud rely on both
the access token and the source code itself. Without access to both, you can't access the cache.

However, this did present the following problem: revoking someone's access to the cache became difficult as long as they
had the source code. Imagine a scenario where someone has left an organization. As long as they have a clone of the
repo, they have everything they need to access the cache, even if their credentials have been
revoked. To fully revoke
their access would require cycling the access token, which could interrupt the work of other developers and CI
pipelines.

## Access Tokens become CI Access Tokens

What we previously called "access tokens" will now be called "[CI access tokens](/ci/recipes/security/access-tokens)."
They are still defined at the workspace

> level, but are designed for use in CI. These tokens can now be set as environment variables on your CI platform so that

Reviewer: ```suggestion
level, but are designed for use in CI. These tokens should be set as environment variables or secrets on your CI platform so that
```

they're no longer committed to your repo.

## What are personal access tokens?

[Personal access tokens](/ci/recipes/security/personal-access-tokens) are a new type of access token that is scoped to
an individual user. This means that this token lives and dies with that member's access to your Nx Cloud workspace.

> Users must log in to Nx Cloud and confirm their access to a workspace before a personal access token can be created.

Reviewer: ```suggestion
Users must log in to Nx Cloud and confirm they are a member of a workspace before a personal access token can be created.
``` 

Just seems off to confirm access before creating an access token?

> Once created, we validate that token for access each time. As soon as a user loses access, the personal access token no

Reviewer: ```suggestion
Once created, we validate that token for access each time you use nx-cloud locally on your machine. As soon as a user loses access, the personal access token no
```

Don't really have strong opinions other than the fact that we should be more specific than `each time`. Each time of what?

longer works and access to the cache is removed.

This gets even more powerful when combined with the GitHub VCS integration. When a user's GitHub access is removed from
a GitHub-connected organization, their access to Nx Cloud is removed, and their personal access token is invalidated.
This means that Nx Cloud can fit into existing user de-provisioning processes you already have.

Open source teams also benefit from personal access tokens. You can configure your access to allow anonymous users to
read from the cache, but limit read/write access to core contributors.

## Controlling default access

![Personal access token settings in Nx Cloud workspace](/blog/images/2024-09-05/workspace-settings.avif)

By default, a workspace that opts in to personal access tokens will allow anonymous users (users without a personal
access token defined) read-only access. This can be changed to disallow access to the cache for anonymous users
in your workspace settings.

> User with personal access tokens will have read-only access to the cache. This can be changed to enable read-write

Reviewer: ```suggestion
Users with personal access tokens will have read-only access to the cache. This can be changed to enable read-write
```
